impl/src/main/java/org/jboss/weld/bean/InterceptorImpl.java

```diff
@@ -125,19 +125,29 @@ public void initializeAfterBeanDiscovery() {
     }
 
     private void checkInterceptorBindings() {
-        if (interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_CONSTRUCT) ||
-                interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_DESTROY) ||
-                interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_ACTIVATE) ||
-                interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_PASSIVATE)) {
+        if (interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_CONSTRUCT)
+                || interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_DESTROY)
+                || interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_ACTIVATE)
+                || interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_PASSIVATE)) {
             for (Annotation interceptorBindingType : interceptorBindingTypes) {
                 Target target = interceptorBindingType.annotationType().getAnnotation(Target.class);
-                if (target != null && Arrays2.unorderedEquals(target.value(), ElementType.TYPE, ElementType.METHOD)) {
-                    throw ReflectionLogger.LOG.methodElementTypeNotAllowed(this, interceptorBindingType.annotationType());
+                if (target == null || hasInvalidTargetType(target.value())) {
```

Comment:
Well, it's not only the case of missing `@Target`. Any other combination than {TYPE, METHOD} was previously ignored. I think it would be quite inconsistent to treat {TYPE, METHOD} differently than {TYPE, METHOD,CONSTRUCTOR} which is also a legal set of valid target types for an interceptor binding. Given that this is not a core functionality but rather a rare corner case which might confuse a lot of users I am in favor of breaking backward compatibility here.